File: Applications/Python/Stack/ReversePolishNotation.py
class Solution:    
    def evalRPN(self, tokens: List[str]) -> int:
        numbers = []        

        for token in tokens:            
            if token == '/':                
                s = numbers.pop()
                f = numbers.pop()
                numbers.append(int(f / s))
            elif token == '*':
                s = numbers.pop()
                f = numbers.pop()
                numbers.append(f * s)                
            elif token == '-':
                s = numbers.pop()
                f = numbers.pop()
                numbers.append(f - s)                
            elif token == '+':
                s = numbers.pop()
                f = numbers.pop()
                numbers.append(f + s)                
            else:                
                numbers.append(int(token))

        return numbers.pop()

File: Applications/CSharp/Stack/ReversePolishNotation.cs
class Solution
{
    public int EvalRPN(string[] tokens)
    {
        Stack<int> stack = new Stack<int>();

        foreach (string token in tokens)
        {
            if (IsNumeric(token))
            {
                stack.Push(int.Parse(token));
            }
            else
            {
                int op2 = stack.Pop();
                int op1 = stack.Pop();
                int result = PerformOperation(token, op1, op2);
                stack.Push(result);
            }
        }

        return stack.Pop();
    }

    private bool IsNumeric(string token)
    {
        return int.TryParse(token, out _);
    }

    private int PerformOperation(string token, int operand1, int operand2)
    {
        switch (token)
        {
            case "+":
                return operand1 + operand2;
            case "-":
                return operand1 - operand2;
            case "*":
                return operand1 * operand2;
            case "/":
                return operand1 / operand2;
            default:
                throw new ArgumentException("Invalid operator: " + token);
        }
    }
}
